onmt/modules/Transformer.py

```diff
@@ -46,15 +46,15 @@ def forward(self, x):
 
 
 class TransformerEncoder(nn.Module):
-    def __init__(self, hidden_size, opt, pad,
+    def __init__(self, hidden_size, dropout, pad,
```

Comment:
This is the only place we need the padding index. See if you can get rid of it somehow.